### ISSUE TITLE
Changing the outgoing message language does not update scheduled messages in real time

### ISSUE THREAD

**Issue by sglangevin** [medic/medic-webapp#4514]
To reproduce:
1) Register a pregnancy using the web app. See that scheduled messages are generated.
2) Change the default language for outgoing messages
3) Notice that scheduled messages are still in their original language

Expected behavior is the same as the behavior for phone number updates - should be updated when the report is loaded again. I tested with translation keys throughout app_settings and with translation keys everywhere except for the notifications (on/off forms).

**Comment by garethbowen** [comment]
@sglangevin I think this is working as expected. We determine the locale in this order:

1. The `locale` property on the doc.
2. The `locale` property on the `doc.sms_message`.
3. The configured default outgoing message locale.
4. The configured default locale.
5. English.

In the examples I can find on beta.dev the `doc.sms_message.locale` is set to "en" so the configuration is never used. You didn't link a doc so I can't tell for sure.

I'm fairly sure this hasn't changed recently, however I think our use cases are changing so messaging someone other than the sender is becoming more common so we could consider being more sophisticated about working out the locale. For example, we should probably only use the locale of the incoming message if we're replying to the sender. To be even more useful we could store the language of the person on the person doc and then use that to set the language. However my understanding is in practice all our projects use a single language so I'm not sure how useful this would be.

**Comment by isaacholeman** [comment]
Just wanted to clarify here @garethbowen @sglangevin, is it the plan to store a contact's preferred language in their profile and be able to configure message schedules that send in the contact's language of preference? If so, we would definitely use this for zazic-zimbabwe. We're currently doing this kind of manually by having two separate message schedules and enrolling them in either the Shona or the English, but it sort of breaks in that auto-responses all have to be in the same language. For the time being we've just shortened our auto-responses so that we could put both languages in the same SMS.

**Comment by sglangevin** [comment]
@isaacholeman no plan is in place yet, but your input is helpful, thanks!

@garethbowen I'll respond to the easy part first 😀. We definitely have a need to have outgoing SMS be sent in multiple languages from the same instance. This has come up mainly in Nepal where we are going to work in districts in which there may be multiple local languages, but it's also useful for other projects. A related issue is that once we start sending outgoing messages in many languages, the person using the web app might not actually be able to understand the messages in the UI, so it may be useful for them to have the option to see a version that is translated into the language of their logged-in user. It makes it harder to do testing but I've heard this request as well.

As far as the issue reported here, I realized my testing was flawed because medic-reporter is hard-coding the locale. I think we should removed that feature from medic-reporter as it creates a scenario that doesn't exist in real life. I'm not sure then if I can actually test the new feature where updating the language should update the language for outgoing messages, which was part of #3412. What use case did you have in mind?

**Comment by garethbowen** [comment]
@sglangevin 

> We definitely have a need to have outgoing SMS be sent in multiple languages from the same instance.

I agree but I think this is a feature request rather than a bug - can we raise this and schedule it separately?

> I think we should removed that feature from medic-reporter

I've added this to #4516

> I'm not sure then if I can actually test the new feature where updating the language should update the language for outgoing messages

I'm not sure how comfortable you feel about this but one way to test it is to delete the `sms_message.locale` property off the doc in futon. However when I did this it still didn't work and it looks like there is a bug where the [message utils calls the translate function](https://github.com/medic/medic-webapp/blob/2.15.x/static/js/modules/message-utils.js#L248) with args that don't match [the expected](https://github.com/medic/medic-webapp/blob/2.15.x/static/js/services/format-data-record.js#L242). I'll get on to this bit today.

**Comment by garethbowen** [comment]
Code review please @dianabarsan . This needs to be backported to `2.15.x`.

**Comment by sglangevin** [comment]
Thanks @garethbowen - we can definitely file that as a separate issue as it's clearly a feature request 😁

Glad you were able to find & fix a bug in the meantime!

**Comment by dianabarsan** [comment]
Merged into `master` and backported to `2.15.x`.

**Comment by ngaruko** [comment]
Seems to pick up the change of language for outgoing messages. However, I am observing a strange behavior: when switching to another language, the top report outgoing messages are not translated at first, until I navigate to another report and come back...
![image](https://user-images.githubusercontent.com/6979995/40596317-276467f0-628e-11e8-8656-49b9457bd42f.png)
Also (maybe not entirely related but worth mentioning here), when changing the user's language in `Edit user profile`, the app does not show the `updates available` modal for the changes to take effect if the user has another tab/browser open. This will happen only after reloading the page.

**Comment by sglangevin** [comment]
@ngaruko does this pass AT then?

**Comment by sglangevin** [comment]
I gave this a quick test on 2.15.0-beta.15 and it seems to be working well. I'm going to move this one to ready. @ngaruko please re-open if you disagree.
